File: src/net/machinemuse/numina/basemod/Numina.scala
package net.machinemuse.numina.basemod

import cpw.mods.fml.common.{SidedProxy, Mod}
import cpw.mods.fml.common.network.{NetworkRegistry, NetworkMod}
import cpw.mods.fml.common.event._
import net.machinemuse.numina.command.Commander
import net.machinemuse.numina.network.{NuminaPackets, MusePacketHandler}
import java.io.File
import net.machinemuse.numina.recipe.JSONRecipeList
import cpw.mods.fml.common.registry.GameRegistry

/**
 * Author: MachineMuse (Claire Semple)
 * Created: 6:06 AM, 6/18/13
 */
@Mod(modid = "numina", modLanguage = "scala")
@NetworkMod(clientSideRequired = true, serverSideRequired = false, tinyPacketHandler = classOf[MusePacketHandler])
object Numina {
  @SidedProxy(clientSide = "net.machinemuse.numina.basemod.NuminaProxyClient", serverSide = "net.machinemuse.numina.basemod.NuminaProxyServer")
  var proxy: NuminaProxy = null
  var configDir: java.io.File = null

  @Mod.EventHandler def preinit(e: FMLPreInitializationEvent) {
    NuminaConfig.init(e)
    configDir = e.getModConfigurationDirectory

    val recipesFolder = new File(configDir, "machinemuse/recipes")
    recipesFolder.mkdirs()
    recipesFolder.mkdir()
    //MinecraftForge.EVENT_BUS.register(PlayerTickHandler)
    //    MinecraftForge.EVENT_BUS.register(DeathEventHandler)
    //    NetworkRegistry.instance.registerGuiHandler(Numina, NuminaGuiHandler)
    proxy.PreInit()
  }

  @Mod.EventHandler def init(e: FMLInitializationEvent) {
    proxy.Init()
    NuminaPackets.init()
  }

  @Mod.EventHandler def postinit(e: FMLPostInitializationEvent) {
    proxy.PostInit()
  }

  @Mod.EventHandler def serverstart(e: FMLServerStartedEvent) {
    Commander.init()
    JSONRecipeList.loadRecipesFromDir(Numina.configDir.toString + "/machinemuse/recipes/")
    GameRegistry.registerPlayerTracker(NuminaPlayerTracker)
    NetworkRegistry.instance.registerConnectionHandler(NuminaConnectionTracker)
  }
}

File: src/main/java/net/machinemuse/numina/basemod/Numina.java
package net.machinemuse.numina.basemod;

import cpw.mods.fml.common.FMLCommonHandler;
import cpw.mods.fml.common.Mod;
import cpw.mods.fml.common.SidedProxy;
import cpw.mods.fml.common.event.FMLInitializationEvent;
import cpw.mods.fml.common.event.FMLPostInitializationEvent;
import cpw.mods.fml.common.event.FMLPreInitializationEvent;
import cpw.mods.fml.common.event.FMLServerStartedEvent;
import net.machinemuse.numina.basemod.proxy.CommonProxy;
import net.machinemuse.numina.recipe.JSONRecipeList;

import javax.annotation.Nonnull;
import java.io.File;

import static net.machinemuse.numina.basemod.Numina.MODID;
import static net.machinemuse.numina.basemod.Numina.VERSION;

/**
 * Author: MachineMuse (Claire Semple)
 * Created: 6:06 AM, 6/18/13
 *
 * Ported to Java by lehjr on 11/15/16.
 */
@Mod(modid = MODID, version = VERSION)
public class Numina {
    public static final String MODID = "numina";
    public static final String VERSION = "@numina_version@";

    @SidedProxy(clientSide = "net.machinemuse.numina.basemod.proxy.ClientProxy", serverSide = "net.machinemuse.numina.basemod.proxy.ServerProxy")
    static CommonProxy proxy;
    public static File configDir = null;

    @Nonnull
    private static Numina INSTANCE;

    @Nonnull
    @Mod.InstanceFactory
    public static Numina getInstance() {
        if (INSTANCE == null)
            INSTANCE = new Numina();
        return INSTANCE;
    }

    @Mod.EventHandler
    private void preinit(FMLPreInitializationEvent event) {
        proxy.preInit(event);
    }

    @Mod.EventHandler
    private void init(FMLInitializationEvent event) {
        proxy.init(event);
    }

    @Mod.EventHandler
    private void postinit(FMLPostInitializationEvent event) {
        proxy.postInit(event);
    }

    @Mod.EventHandler
    private void serverstart(FMLServerStartedEvent event) {
        JSONRecipeList.loadRecipesFromDir(Numina.getInstance().configDir.toString() + "/machinemuse/recipes/");
        FMLCommonHandler.instance().bus().register(new NuminaPlayerTracker());
    }
}